Objective-C/CBLCollection.h

```diff
@@ -19,21 +19,59 @@
 
 #import <Foundation/Foundation.h>
 #import "CBLDatabase.h"
-#import "CBLDatabaseChangeObservable.h"
+#import "CBLCollectionChangeObservable.h"
 #import "CBLIndexable.h"
-
-@protocol CBLListenerToken;
-@class CBLScope;
 @class CBLDocument;
-@class CBLMutableDocument;
 @class CBLDocumentChange;
+@class CBLMutableDocument;
+@class CBLScope;
+@protocol CBLListenerToken;
 
 NS_ASSUME_NONNULL_BEGIN
 
 /** The default collection name constant */
 extern NSString* const kCBLDefaultCollectionName;
 
-@interface CBLCollection : NSObject<CBLDatabaseChangeObservable, CBLIndexable>
+/**
+ A `CBLCollection` represent a collection which is a container for documents.
+ 
+ A collection can be thought as a table in the relational database. Each collection belongs to
+ a scope which is simply a namespce, and has a name which is unique within its scope.
+ 
+ When a new database is created, a default collection named "_default" will be automatically
+ created. The default collection is created under the default scope named "_default".
+ You may decide to delete the default collection, but noted that the default collection cannot
+ be re-created. The name of the default collection and scope can be referenced by using
+ `kCBLDefaultCollectionName` and `kCBLDefaultScopeName` constant.
+ 
+ When creating a new collection, the collection name, and the scope name are required.
+ The naming rules of the collections and scopes are as follows:
+ - Must be between 1 and 251 characters in length.
+ - Can only contain the characters A-Z, a-z, 0-9, and the symbols _, -, and %.
+ - Cannot start with _ or %.
+ - Both scope and collection names are case sensitive.
+ 
+ ##  `CBLCollection` Lifespan
+ `CBLCollection` is ref-counted and is owned by the database object that creates it. Hence,
```

Comment:
This is not related to Objective-C as user will not do call retain() or release(). Can you revise this?